addons/web_editor/tools.py

```diff
@@ -0,0 +1,133 @@
+    ''' Computes the platform name and embed_url from given URL
+        (or error message in case of invalid URL).
+    '''
+    source = get_video_source_data(video_url)
+    if not source:
+        return {'error': True, 'message': _('The provided url is invalid')}
+
+    autoplay = kwargs.get('autoplay') and '?autoplay=1&mute=1' or '?autoplay=0'
+    controls = kwargs.get('hide_controls') and '&controls=0' or ''
+    loop = kwargs.get('loop') and '&loop=1' or ''
+
+    # We directly use the provided URL as it is
+    embed_url = video_url
+    platform = source[0]
+    platform_id = source[1]
+    platform_match = source[2]
+    if platform == 'youtube':
+        fullscreen = kwargs.get('hide_fullscreen') and '&fs=0' or ''
+        youtube_loop = loop and loop + '&playlist=%s' % (platform_id) or ''
+        logo = kwargs.get('hide_yt_logo') and '&modestbranding=1' or ''
+        params = '%s&rel=0%s%s%s%s' % (autoplay, youtube_loop, controls, fullscreen, logo)
+        yt_extra = platform_match.groups()[0] or ''
+        embed_url = f'//www.youtube{yt_extra}.com/embed/{platform_id}{params}'
+    elif platform == 'vimeo':
+        vimeo_autoplay = autoplay.replace('mute', 'muted')
+        embed_url = f'//player.vimeo.com/video/{platform_id}{vimeo_autoplay}{loop}'
+    elif platform == 'dailymotion':
+        logo = kwargs.get('hide_dm_logo') and '&ui-logo=0' or ''
+        share = kwargs.get('hide_dm_share') and '&sharing-enable=0' or ''
+        params = f'{autoplay}{controls}{logo}{share}'
+        embed_url = f'//www.dailymotion.com/embed/video/{platform_id}{params}'
+    elif platform == 'instagram':
+        embed_url = f'//www.instagram.com/p/{platform_id}/embed/'
+    elif platform == 'youku':
+        embed_url = f'//player.youku.com/embed/{platform_id}'
+
+    return {'platform': platform, 'embed_url': embed_url}
+
+
+def get_video_embed_code(video_url):
+    ''' Computes the valid iframe from given URL that can be embedded
+        (or False in case of invalid URL).
+    '''
+    data = get_video_url_data(video_url)
+    if data.get('error'):
+        return False
+    return Markup('<iframe class="embed-responsive-item" src="%s" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowFullScreen="true" frameborder="0"></iframe>') % data.get('embed_url')
+
+
+def get_video_thumbnail(video_url):
+    ''' Computes the valid thumbnail image from given URL
+        (or False in case of invalid URL).
+    '''
+    source = get_video_source_data(video_url)
+    if not source:
+        return False
+
+    response = False
+    platform = source[0]
+    platform_id = source[1]
+    if platform == 'youtube':
+        response = requests.get(f'https://img.youtube.com/vi/{platform_id}/0.jpg', timeout=10)
+    elif platform == 'vimeo':
+        req = requests.get(f'http://vimeo.com/api/oembed.json?url={video_url}', timeout=10)
+        if req.status_code == 200:
+            data = json.loads(req.content)
```

Comment:
That's built-in:
```suggestion
            data = req.json()
```
also should  be `res` rather than `req`, it's not a request.